src/core/messages.ts

```diff
@@ -200,89 +280,328 @@ export interface MessageSubscriptionResponse extends Subscription {
  */
 export interface Messages {
   /**
-   * Subscribe to new messages in this chat room.
-   * @param listener callback that will be called
-   * @returns A response object that allows you to control the subscription.
+   * Subscribe to chat message events in this room.
+   *
+   * This method allows you to listen for chat message events and provides access to
+   * historical messages that occurred before the subscription was established.
+   *
+   * **Note**: The room must be attached for the listener to receive new message events.
+   * @param listener - A callback function that will be invoked when chat message events occur.
+   * @returns A {@link MessageSubscriptionResponse} object that provides:
+   *          - `unsubscribe()`: Method to stop listening for message events
+   *          - `historyBeforeSubscribe()`: Method to retrieve messages sent before subscription
+   * @example
+   * ```typescript
+   * import * as Ably from 'ably';
+   * import { ChatClient, ChatMessageEventType } from '@ably/chat';
+   *
+   * const chatClient: ChatClient; // existing ChatClient instance
+   *
+   * // Get a room and subscribe to messages
+   * const room = await chatClient.rooms.get('general-chat');
+   *
+   * const subscription = room.messages.subscribe((event) => {
+   *   console.log(`Message ${event.type}:`, event.message.text);
+   *   console.log('From:', event.message.clientId);
+   *   console.log('At:', event.message.timestamp);
+   *
+   *   // Handle different event types
```

Comment:
we print `event.type`. is that enough for the example without this switch as well?